discovery_engine_core/providers/src/gnews.rs

```diff
@@ -12,52 +12,231 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use chrono::{DateTime, Utc};
+//! Client to get new documents.
```

Comment:
❓ is there a specific reason to move the client related code to the parent module? instead of a module tree like
```
- gnews
  - models
```
i would have expected sth like
```
- gnews
  - client
  - models
```
or if we want to be more modular even sth like
```
- gnews
  - client
  - models
  - query
```
similarly for the newscatcher modules.

Reply:
I guess it's a stylistic question, but I don't see the use for it as it only creates an "empty" gnews module which does nothing else but reexport the submodules.

Now putting this in context of the follow up PR:

- queries will be the same for all providers so there is no point for a `<provider>/query` module.
- models will be a purely internal implementation detail of the `gnews` provider
- everything else is the entry points of the module, we could move all of that into client, but I don't see any benefits in doing so, I mean it's like moving the whole main code of a module into a sub-module just because there are other sub-modules IMHO.